content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md

```diff
@@ -0,0 +1,32 @@
+
+Regardless of whether you are rolling your own cluster or using a managed Kubernetes service from a cloud provider, you may need to [migrate telemetry or security agents that rely on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/). 
+
+## I have a Docker dependency. What now?
+
+If your Kubernetes cluster does depend on Docker Engine and you intend to upgrade to Kubernetes v1.24 (which you should eventually do for security and similar reasons), you will need to either change your container runtime from Docker Engine to something else, or use [cri-dockerd](https://github.com/Mirantis/cri-dockerd). Since [containerd](https://containerd.io/) is a graduated CNCF project and the runtime within Docker itself, it’s a safe bet as an alternative container runtime. Fortunately, the Kubernetes project has already documented the process of [changing a node’s container runtime](/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/), using containerd as the example. Instructions are similar for switching to one of the other supported runtimes.
+
+## I want to upgrade Kubernetes, and I need to maintain compatibility with Docker as a runtime. What are my options?
+
+Fear not, you aren’t being left out in the cold and you don’t have to take the security risk of staying on an old version of Kubernetes. Mirantis and Docker have jointly released, and are maintaining, a replacement for dockershim. That replacement is called [cri-dockerd](https://github.com/Mirantis/cri-dockerd). If you do need to maintain compatibility with Docker as a runtime, install cri-dockerd following the instructions in the project’s documentation.
+
+## Is that it?
+
+
+Yes. As long as you go into this release aware of the changes being made and the details of your own clusters, as well as communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, your application code, or some scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman, since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run. If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```

Comment:
```suggestion
Yes. As long as you go into this release aware of the changes being made and the details of your own clusters and communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, application code, or scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run. If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```

Reply:
For clarity, maybe this:
```suggestion
Yes. As long as you go into this release aware of the changes being made and the details of your own clusters, and you make sure to communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, application code, or scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run.

If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```

Reply:
Do you think it makes sense to specifically call out containerd here? I would like to see us be less opinionated about which supported runtime we advocate and instead direct people to the list of supported options: https://kubernetes.io/docs/setup/production-environment/container-runtimes/.